tests/test_lens.py

```diff
@@ -373,14 +430,85 @@ def test_image_number(self):
+        assert (image_number[0] == 4) or (image_number[0] == 2) or (image_number[0] == 1)
+
+        gg_lens_multisource = Lens(
+            source_class=[self.source6, self.source6],
+            deflector_class=self.deflector6,
+            cosmo=self.cosmo,
+            los_config=los,
+        )
+        kwargs_model = gg_lens_multisource.lenstronomy_kwargs()[0]
+        kwargs_model_keys = kwargs_model.keys()
+        expected_kwargs_model = ['lens_light_model_list',
+                                'lens_model_list',
+                                'z_lens',
+                                'lens_redshift_list',
+                                'source_redshift_list',
+                                'z_source_convention',
+                                'cosmo',
+                                'source_light_model_list']
+        assert expected_kwargs_model[0] in kwargs_model_keys
+        assert expected_kwargs_model[1] in kwargs_model_keys
+        assert expected_kwargs_model[2] in kwargs_model_keys
+        assert expected_kwargs_model[3] in kwargs_model_keys
+        assert expected_kwargs_model[4] in kwargs_model_keys
+        assert expected_kwargs_model[5] in kwargs_model_keys
+        assert expected_kwargs_model[6] in kwargs_model_keys
+
+@pytest.fixture
+def supernovae_lens_instance_double_sersic_multisource():
+    path = os.path.dirname(__file__)
+    source_dict = Table.read(
+        os.path.join(path, "TestData/source_supernovae_new.fits"), format="fits"
+    )
+    deflector_dict = Table.read(
+        os.path.join(path, "TestData/deflector_supernovae_new.fits"), format="fits"
+    )
+
+    cosmo = FlatLambdaCDM(H0=70, Om0=0.3)
+    while True:
+        source = Source(
+            source_dict=source_dict,
+            cosmo=cosmo,
+            source_type="point_plus_extended",
+            light_profile="double_sersic",
+            lightcurve_time=np.linspace(-20, 100, 1000),
+            variability_model="light_curve",
+            kwargs_variability={"supernovae_lightcurve", "i"},
+            sn_type="Ia",
+            sn_absolute_mag_band="bessellb",
+            sn_absolute_zpsys="ab",
+
+        )
+        deflector = Deflector(
+                deflector_type="EPL",
+                deflector_dict=deflector_dict,
+            )
+        supernovae_lens = Lens(
+            deflector_class=deflector,
+            source_class=[source, source],
+            cosmo=cosmo,
+        )
+        if supernovae_lens.validity_test():
+            supernovae_lens = supernovae_lens
+            break
+    return supernovae_lens
+
+def test_double_sersic_multisource(supernovae_lens_instance_double_sersic_multisource):
+    lens_class = supernovae_lens_instance_double_sersic_multisource
+    results = lens_class.source_light_model_lenstronomy(band="i")
+    assert len(results[0]["source_light_model_list"]) == 2
+    assert len(results[1]["kwargs_source"]) == 2
+    assert len(results[1]["kwargs_ps"]) == 2
 
 
+        
```

Comment:
I'd like to see the following test for the Lens() class:
1. Initiate two Lens() classes with each the same deflector and different point+extended sources at two different redshift
2. Initiate a third Lens() class with the same deflector and both sources at the same time
3. Then check that all the properties, such as time delays, magnifications, extended source magnifications etc are identical between the the separate and the combined classes.